src/zope/testrunner/runner.py

```diff
@@ -451,8 +451,12 @@ def run_layer(options, layer_name, layer, tests, setup_layers,
         raise
     except Exception:
         f = StringIO()
-        traceback.print_exc(file=f)
-        output.error(f.getvalue())
+        try:
+            traceback.print_exc(file=f)
+            output.error(f.getvalue())
+        except Exception:
+            output.error(
+                '*** An exception occurred, but could not be formatted ***')
```

Comment:
It would be useful to see (a) the name and str() of the original exception, and (b) the name and str() of the exception that occurred while trying to format the original exception.